ruby/ql/lib/codeql/ruby/controlflow/internal/ControlFlowGraphImpl.qll

```diff
@@ -1365,6 +1365,10 @@ module Trees {
     final override ControlFlowTree getChildElement(int i) { result = this.getComponent(i) }
   }
 
+  private class StringComponentComponentTree extends LeafTree, StringComponent {
+    StringComponentComponentTree() { not this instanceof StringInterpolationComponent }
```

Comment:
Are interpolations included in the CFG elsewhere?

Reply:
I don't think so, but perhaps they should. But I would rather that we do that in a separate PR, since adding them naively makes the CFG inconsistent (some nodes will have multiple successors).

Reply:
Why is this called `StringComponentComponentTree` rather than `StringComponentTree`?

Reply:
That's a typo.